### PR TITLE
Add ramen cloud credentials secret in managed clusters

### PR DESCRIPTION
**Pull request by nirs**
We configured ramen to access velero cloud-credentials secret:

    veleroNamespaceSecretKeyRef:
      key: cloud
      name: cloud-credentials

Using the secret created by `velero install`. However this secret is installed in `velero` namespace and we need it in `ramen-system` namespace.

### CODE REVIEW

File: ramenctl/ramenctl/config.py [hunk 1]
```diff
@@ -98,3 +98,32 @@ def run(args):
...
     )
+
+    # NOTE: Need to ba last since ramen namespace on the managed clusters is
+    # created by ramen in previous steps.
+
+    template = drenv.template(command.resource("cloud-credentials-secret.yaml"))
+    for cluster in env["clusters"]:
+        command.info("Waiting for ramen-system namespace in cluster '%s'", cluster)
+        drenv.wait_for(
+            "namespace/ramen-system",
```
Comment (nirs) [inline ramenctl/ramenctl/config.py:109]: Need to use args.ramen_namespace

File: ramenctl/ramenctl/resources/cloud-credentials-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: cloud-credentials
+  namespace: ramen-system
```
Comment (nirs) [inline ramenctl/ramenctl/resources/cloud-credentials-secret.yaml:10]: Need to use args.ramen_namespace instead of hardcoding the value

**Comment by nirs** [comment]
Secrets in cluster after `ramenctl config`:

velero secret (created by velero install):
```
$ kubectl get secret/cloud-credentials -n  velero -o jsonpath='{.data.cloud}' | base64 -d
[default]
aws_access_key_id = minio
aws_secret_access_key = minio123
```

ramen secret (created by ramenctl config):
```
$ kubectl get secret/cloud-credentials -n ramen-system -o jsonpath='{.data.cloud}' | base64 -d
[default]
aws_access_key_id = minio
aws_secret_access_key = minio123
```

File: ramenctl/ramenctl/config.py [hunk 3]
```diff
@@ -32,13 +32,36 @@ def run(args):
...
+        "secret/cloud-credentials",
+        "--namespace=velero",
+        "--output=jsonpath={.data.cloud}",
+        context=env["clusters"][0],
+    )
+    template = drenv.template(command.resource("cloud-credentials-secret.yaml"))
+    yaml = template.substitute(cloud=cloud, namespace=args.ramen_namespace)
+
+    for cluster in env["clusters"]:
+        command.info("Creating cloud credentials secret in clsuter '%s'", cluster)
```
Comment (ELENAGER) [inline ramenctl/ramenctl/config.py:57]: Typo - cluster

Comment (nirs) [reply ramenctl/ramenctl/config.py]: Fixed in current version.

File: ramenctl/ramenctl/unconfig.py [hunk 4]
```diff
@@ -39,10 +39,24 @@ def run(args):
...
+        input=template.substitute(namespace=args.ramen_namespace),
         context=env["hub"],
         log=command.debug,
     )
+
+    template = drenv.template(command.resource("cloud-credentials-secret.yaml"))
+    yaml = template.substitute(cloud="", namespace=args.ramen_namespace)
+
+    for cluster in env["clusters"]:
+        command.info("Deleting cloud credentials secret in clsuter '%s'", cluster)
```
Comment (raghavendra-talur) [inline ramenctl/ramenctl/unconfig.py:55]: I don't think the typo is fixed. Did you push the update?

**Review by raghavendra-talur — CHANGES_REQUESTED** [review]

File: ramenctl/ramenctl/unconfig.py (hunk 4 above)
Comment (raghavendra-talur) [reply ramenctl/ramenctl/unconfig.py:55]: Oh, this is in a different file. Please update this file too. Rest of the changes look good.

Comment (nirs) [reply ramenctl/ramenctl/unconfig.py:55]: Fixing, thanks!

**Review by ELENAGER — APPROVED** [review]

**Comment by nirs** [comment]
@raghavendra-talur current version fixed the typo in the log and is ready for merge.

**Review by tjanssen3 — APPROVED** [review]
I think creating a single secret and then copying it over is the correct approach for easier maintenance later. LGTM.

**Pull request merged**